modules/enum_ec2/main.py

```diff
@@ -1,483 +1,482 @@
+            response = client.describe_subnets()
+            for subnet in response['Subnets']:
+                subnet['Region'] = region
+                subnets.append(subnet)
+            print('  {} subnet(s) found.'.format(len(subnets)))
+            all_subnets += subnets
+
+        # VPCs
+        if args.vpcs is True or all is True:
+            response = client.describe_vpcs()
+            for vpc in response['Vpcs']:
+                vpc['Region'] = region
+                vpcs.append(vpc)
+            print('  {} VPC(s) found.'.format(len(vpcs)))
+            all_vpcs += vpcs
+
+        # VPC Endpoints
+        if args.vpc_endpoints is True or all is True:
+            response = None
+            next_token = False
+            while (response is None or 'NextToken' in response):
+                if next_token is False:
+                    response = client.describe_vpc_endpoints(
+                        MaxResults=1000
+                    )
+                else:
+                    response = client.describe_vpc_endpoints(
+                        NextToken=next_token,
+                        MaxResults=1000
+                    )
+                if 'NextToken' in response:
+                    next_token = response['NextToken']
+                for endpoint in response['VpcEndpoints']:
+                    endpoint['Region'] = region
+                    vpc_endpoints.append(endpoint)
+            print('  {} VPC endpoint(s) found.'.format(len(vpc_endpoints)))
+            all_vpc_endpoints += vpc_endpoints
+
+        print('')  # Break the line after each region. This isn't on the end of another print because they won't always be all used and isn't before the region print because it would double break lines at the beginning
+
+    gathered_data = {
+        'Instances': all_instances,
+        'SecurityGroups': all_security_groups,
+        'ElasticIPs': all_elastic_ips,
+        'VPNCustomerGateways': all_vpn_customer_gateways,
+        'DedicatedHosts': all_dedicated_hosts,
+        'NetworkACLs': all_network_acls,
+        'NATGateways': all_nat_gateways,
+        'NetworkInterfaces': all_network_interfaces,
+        'RouteTables': all_route_tables,
+        'Subnets': all_subnets,
+        'VPCs': all_vpcs,
+        'VPCEndpoints': all_vpc_endpoints,
+    }
+
+    ec2_data = deepcopy(session.EC2)
+    for key, value in gathered_data.items():
+        ec2_data[key] = value
+    session.update(pacu_main.database, EC2=ec2_data)
+
+    # Add regions to gathered_data for summary output
+    gathered_data['regions'] = regions
+
+    print('All data has been saved to the current session.\n')
+
+    print('{} completed.\n'.format(module_info['name']))
+
+    return gathered_data
+
+
+def summary(data, pacu_main):
+    results = []
+
+    results.append('  Regions:')
+    for region in data['regions']:
+        results.append('     {}'.format(region))
+
+    results.append('')
+
+    if len(data['Instances']) > 0:

```

Comment:
This is a hard problem to solve with how summary is setup but we still want to show when 0 of something was found, we only want to hide things that weren't even attempted for enumerartion